silkrpc/http/request_handler.cpp

```diff
@@ -47,18 +47,48 @@ boost::asio::awaitable<void> RequestHandler::handle_request(const http::Request&
         SILKRPC_DEBUG << "handle_request content: " << request.content << "\n";
 
         const auto request_json = nlohmann::json::parse(request.content);
-        if (!request_json.contains("id")) {
-            reply.content = "\n";
-            reply.status = http::StatusType::ok;
-        } else {
-            auto request_id = request_json["id"].get<uint32_t>();
-            const auto error = co_await is_request_authorized(request_id, request);
-            if (error.has_value()) {
-                reply.content = make_json_error(request_id, 403, error.value()).dump() + "\n";
-                reply.status = http::StatusType::unauthorized;
+
+        if (request_json.is_object()) {
+            if (!request_json.contains("id")) {
+                reply.content = "\n";
+                reply.status = http::StatusType::ok;
             } else {
-                co_await handle_request(request_json, reply);
+                auto request_id = request_json["id"].get<uint32_t>();
+                const auto error = co_await is_request_authorized(request_id, request);
+                if (error.has_value()) {
+                    reply.content = make_json_error(request_id, 403, error.value()).dump() + "\n";
+                    reply.status = http::StatusType::unauthorized;
+                } else {
+                    co_await handle_request(request_json, reply, /* is_batch */ false);
```

Comment:
The new parameter `is_batch` seems to be used just to append one newline at the end of the reply content when the request is not a batch one.
If I'm not missing something, you can obtain the same result in a simpler way:
- remove the `is_batch` parameter
- append one newline at the end of the reply content just after this `handle_request` call

Reply:
Yes it's better